### PR TITLE
extend tabs block

### LINKED ISSUE #129: tab block extension
extend tab block to allow for the following version: 

Screenshot 2024-08-28 at 9 44 00 PM

### PR DESCRIPTION
**Pull request by MarquiseRosier**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #129 

Test URLs:
- Before: https://main--shredit--stericycle.aem.page/en-us/who-we-serve/accountants
- After: 
  - https://tab-extension--shredit--stericycle.aem.page/en-us/secure-shredding-services/shredding-events/community-shred-events-resources
  - https://tab-extension--shredit--stericycle.aem.page/en-us/who-we-serve/accountants
  - https://tab-extension--stericycle-shared--aemsites.aem.page/en-us/who-we-serve/accountants

### CODE REVIEW

**Comment by jindaliiita** [comment]
Original
Screenshot 2024-08-29 at 9 48 49 PM

EDS
Screenshot 2024-08-29 at 9 48 53 PM

Mobile view seems to be different on both

File: blocks/tabs/tabs.css [hunk 1]
```diff
@@ -5,6 +5,33 @@
...
+.tabs.vertical .tabs-list {
+    flex-direction: column;
+    margin-bottom: 3.75rem;
+    -ms-flex-preferred-size: 30%;
+    flex-basis: 30%;
+    flex-shrink: 0;
+    border-right: 2px solid #c0c6c6;
+}
+
+.tabs.vertical .tabs-panel {
```
Comment (jindaliiita) [inline blocks/tabs/tabs.css:21]: gap: 2rem;

File: blocks/tabs/tabs.css [hunk 2]
```diff
@@ -29,6 +56,10 @@
     cursor: initial;
 }
```
Comment (jindaliiita) [inline blocks/tabs/tabs.css:57]: .tabs.vertical .tabs-panel>div:last-of-type {
  flex-basis: 40%
}

.tabs.vertical .tabs-panel>div:first-of-type {
  flex-basis: 60%
}

File: blocks/tabs/tabs.css [hunk 3]
```diff
@@ -5,6 +5,33 @@
...
 
+.tabs.vertical {
+    display: flex;
+}
+
+.tabs.vertical .tabs-list {
+    flex-direction: column;
+    margin-bottom: 3.75rem;
+    -ms-flex-preferred-size: 30%;
```
Comment (jindaliiita) [inline blocks/tabs/tabs.css:15]: is  -ms-flex-preferred-size: really needed

File: blocks/tabs/tabs.css [hunk 4]
```diff
@@ -41,6 +72,25 @@
     overflow: auto;
 }
 
+.tabs.vertical .tabs-panel h3 { 
+    color: #133b5c;
```
Comment (jindaliiita) [inline blocks/tabs/tabs.css:76]: can we pick this from the styles.css ?

**Review by dkuntze — APPROVED** [review]

File: blocks/tabs/tabs.css [hunk 5]
```diff
@@ -1,46 +1,282 @@
-.tabs .tabs-list {
+main section .tabs-container {
+    padding: 32px;
+}
+
+main section .tabs-container > div {
+    padding: 0 12px;
+}
+
+.tabs {
```
Comment (jindaliiita) [inline blocks/tabs/tabs.css:9]: can we please add .block before block name

File: blocks/tabs/tabs.js [hunk 6]
```diff
@@ -5,6 +5,25 @@ function hasWrapper(el) {
   return !!el.firstElementChild && window.getComputedStyle(el.firstElementChild).display === 'block';
 }
 
+function addAccordionAnimation(details) {
```
Comment (jindaliiita) [inline blocks/tabs/tabs.js:8]: copilot is suggesting little bit optimization and code readability via
```
function addAccordionAnimation(details) {
  const summary = details.querySelector('summary');

  summary.addEventListener('click', (event) => {
    event.preventDefault(); // Prevent default behavior of summary

    const isOpen = details.hasAttribute('open');
    details.classList.toggle('open', !isOpen);
    details.classList.toggle('close', isOpen);

    if (isOpen) {
      setTimeout(() => {
        details.removeAttribute('open');
        details.classList.remove('close');
      }, 300); // Match this duration with the CSS animation duration
    } else {
      details.setAttribute('open', '');
    }
  });
}
```

**Comment by jindaliiita** [comment]
@MarquiseRosier Have you verified your changes on the pages which contains horizontal tabs ?

File: blocks/tabs/tabs.css [hunk 7]
```diff
@@ -1,46 +1,282 @@
...
+        max-height: 5000vh;
+    }
+
+    to {
+        padding: .125rem 1rem 1rem;
+        max-height: 0;
+    }
+}
+
+.tab-item.open .tab-item-body {
```
Comment (jindaliiita) [inline blocks/tabs/tabs.css:153]: can we please add .block.tab before these
and can you merge 
.tab-item.close .tab-item-body with this

File: blocks/tabs/tabs.js [hunk 8]
```diff
@@ -27,6 +46,30 @@ export default async function decorate(block) {
       tabpanel.lastElementChild.innerHTML = `<p>${tabpanel.lastElementChild.innerHTML}</p>`;
     }
 
+    // decorate faq item label
+    const label = tabpanelCopy.children[1].querySelector('h3');
+    const summary = document.createElement('summary');
```
Comment (jindaliiita) [inline blocks/tabs/tabs.js:51]: I am not sure if adding custom tags are correct or not from SEO perspective.
Why are we not handling with the classes here

**Pull request merged**